packages/test/test-end-to-end-tests/src/test/SummarizerWithSearch.spec.ts

```diff
@@ -321,10 +323,13 @@ describeCompat(
 			containerRuntime: ContainerRuntime;
 			summaryCollection: SummaryCollection;
 		}): Promise<string> {
+			const latestSummaryRefSeqNum =
+				latestAckedSummary?.summaryOp.referenceSequenceNumber ?? 0;
```

Comment:
Side note: This made me really curious why `latestSummaryRefSeqNum` isn't optional in the real code.  Traced all the way back to instantiating the delta manager from the snapshot being loaded from. 0 makes sense for these tests (and actually DM initializes it to 0, but I didn't look to see which codepaths leave it that way - maybe only for detached container, would be used for generating the attach summary? Dunno)

Reply:
By making `latestSummaryRefSeqNum` always defined, it forces the customer to actually think about the correctness of the generation of their summaries. Also, it makes it way harder to miss a spot. This helps catch issues at compile time rather than runtime.